### PR TITLE
Apparmor fixes

### PR DESCRIPTION
**Pull request by vrothberg**
- don't load/set the aa profile when initializing the container
- add a baseline test to avoid regressing on privileged containers

### CODE REVIEW

File: libpod/container_internal_linux.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ import (
 	"github.com/containers/libpod/pkg/lookup"
 	"github.com/containers/libpod/pkg/resolvconf"
 	"github.com/containers/libpod/pkg/rootless"
-	"github.com/cyphar/filepath-securejoin"
+	securejoin "github.com/cyphar/filepath-securejoin"
```
Comment (TomSweeneyRedHat) [inline libpod/container_internal_linux.go]: did you mean to add 'securejoin' here?  It's fine, but I ask as I don't see it in other changes below.

Comment (vrothberg) [reply libpod/container_internal_linux.go]: VSCode applied the change automatically. It seems that this package is always referenced by `securejoin.ABC()`, so the change seemed reasonable to me as it avoids confusion about `filepath-securejoin` vs `securejoin`.

File: test/test_podman_baseline.sh [hunk 2]
```diff
@@ -504,6 +504,16 @@ EOF
...
+    #Expected to pass (as root wit --privileged).
+    #Note that the profile should not be loaded letting the mount succeed.
+    podman run --privilged --security-opt apparmor=$aaProfile docker.io/library/alpine:latest sh -c "mkdir tmp2; mount --bind tmp tmp2"
+    rc=$?
+    echo -n "root with specified AppArmor profile but --privileged: "
+    if [ $rc == 0 ]; then
+        echo "passed"
+    else
+        echo "failed"
+    fi
```
Comment (TomSweeneyRedHat) [inline test/test_podman_baseline.sh]: This test only gets run "by hand".  I like the addition, but do we need anything in the CI test suite too?

Comment (vrothberg) [reply test/test_podman_baseline.sh]: That's a very good questions. Previously, we could not run the tests as Travis disabled AppArmor in their Ubuntu instances but maybe we can run the tests in Cirrus. @cevich, do you know if we can run AppArmor tests in the Ubuntu Cirrus instances?

File: test/test_podman_baseline.sh [hunk 3]
```diff
@@ -504,6 +504,16 @@ EOF
         echo "failed"
     fi
 
+    #Expected to pass (as root wit --privileged).
+    #Note that the profile should not be loaded letting the mount succeed.
+    podman run --privilged --security-opt apparmor=$aaProfile docker.io/library/alpine:latest sh -c "mkdir tmp2; mount --bind tmp tmp2"
```
Comment (rhatdan) [inline test/test_podman_baseline.sh]: Typo on --privilged.

**Comment by rhatdan** [comment]
We should throw an error if the user specifies

--security-opt apparmor=... and --privileged at the same time.

File: test/test_podman_baseline.sh (hunk 3 above)
Comment (vrothberg) [reply test/test_podman_baseline.sh]: Ouch. Thanks for catching!

**Comment by vrothberg** [comment]
> We should throw an error if the user specifies
> 
> --security-opt apparmor=... and --privileged at the same time.

Do we have such checks in pace for selinux?

**Comment by vrothberg** [comment]
> > We should throw an error if the user specifies
> > --security-opt apparmor=... and --privileged at the same time.
> 
> Do we have such checks in place for selinux?

Just checked, we don't check that for selinux either. Not sure if adding the check will cause regressions for some users.

**Comment by rhatdan** [comment]
Well we could add a warning.  Since --privileged will override any --security-opt selection.  Or any --cap-add/cap-drop selection.

**Comment by vrothberg** [comment]
> Well we could add a warning. Since --privileged will override any --security-opt selection. Or any --cap-add/cap-drop selection.

Done :heavy_check_mark::
```
[libpod (apparmor-fixes)]$ sudo ./bin/podman --log-level=warn create --privileged --security-opt label=foo alpine ls
WARN[0000] setting security options with --privileged has no effect 
7218ea3309a8c5c811b4cd75d58fc8ada7b33f80316ebf0d8019cd3f15309897
```

@rhatdan should be happy now. @baude, @mheon, @giuseppe PTAL.

**Comment by rhatdan** [comment]
LGTM

**Comment by baude** [comment]
lets merge ... @vrothberg can you follow up on CI and apparmor?

/lgtm

**Comment by vrothberg** [comment]
> lets merge ... @vrothberg can you follow up on CI and apparmor?

Thanks! Sure, will do.

File: test/test_podman_baseline.sh (hunk 2 above)
Comment (cevich) [reply test/test_podman_baseline.sh]: Yes, and if it's not working out of the box, we own the images (or a copy of them at least) so should be able to make it work.

**Pull request merged**